### PR TITLE
[CFM-905] custom callbacks

### PR DESCRIPTION
**Pull request by ordavidil**
https://issuetracker.amplexor.com/jira/browse/CFM-905

### CODE REVIEW

**Comment by ordavidil** [comment]
selection_220

selection_219

**Comment by ordavidil** [comment]
@soniCaH 
This one is dependent on #1009

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 1]
```diff
@@ -474,8 +491,77 @@ function c4m_user_og_c4m_cta_block_alter(&$block) {
   $variables = array(
     'button_label' => $types[$membership_request_value]['button_label'],
     'button_icon' => $types[$membership_request_value]['button_icon'],
-    'url' => url('/user'),
+    'url' => url('group/join/' . $group->nid),
```
Comment (soniCaH) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: Please discard the purl modifier so we don't have the group navigation and stuff showing in this context. `$options['purl']['disabled']`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/templates/c4m_unified_workflow_register.tpl.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ */
+?>
+<div class="c4m-unified-workflow-title">
+  <h1 class="title">
+    To join the <strong><?php print $group_title; ?></strong> group you need to register with capacity4more by filling in your details below.
+  </h1>
+  <h3 class="login">
+    Already a member? <?php print $login_link; ?>
+  </h3>
+</div>
```
Comment (soniCaH) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/templates/c4m_unified_workflow_register.tpl.php]: screen shot 2016-07-13 at 11 32 02

Few issues here:

1. This tpl is printed within an h1-tag, so we're nesting div/h1/h3 inside an h1-tag.
1. I think an h3 is more then sufficient
1. There's a lot of margin on top of it.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/templates/c4m_unified_workflow_login.tpl.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+/**
+ * @file
+ * Template to render the title of the login form under unified workflow.
+ */
+?>
+<div class="c4m-unified-workflow-title">
+  <h1 class="title">
+    Join the <strong><?php print $group_title; ?></strong> group
+  </h1>
+</div>
```
Comment (soniCaH) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/templates/c4m_unified_workflow_login.tpl.php]: Div + h1 inside h1 here. Please print a single h3, which is big enough.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 4]
```diff
@@ -474,8 +491,77 @@ function c4m_user_og_c4m_cta_block_alter(&$block) {
...
+ */
+function c4m_user_og_form_user_register_form_alter(&$form, &$form_state, $form_id) {
+  if (!isset($_SESSION['gid_join']) || !$gid = $_SESSION['gid_join']) {
+    return;
+  }
+
+  $wrapper = entity_metadata_wrapper('node', $gid);
+  $variables = array(
+    'group_title' => $wrapper->label(),
+    'login_link' => l('log in', 'user/login'),
```
Comment (soniCaH) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: Log in. (please capitalise and add ".")

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module (hunk 1 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: @soniCaH @janlemmens 
with the options I get the same result,
I don't understand your concern, can you please let me know what the url you would like to achieve here?
thanks.

**Pull request merged**